eland/operations.py

```diff
@@ -235,6 +248,10 @@ def _metric_aggs(self, query_compiler, func, field_types=None, numeric_only=None
                     results[field] = elasticsearch_date_to_pandas_date(
                         response["aggregations"][field]["value_as_string"], date_format
                     )
+                elif keep_original_dtype:
```

Comment:
Is there a test case that fails if this behavior isn't present already? If not we should create one to make sure we don't regress.

Reply:
Yes there is a case that fails if this behavior is not present